adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -235,13 +248,7 @@ private void connectionRequestHandler(final ProtonConnection connRequest) {
             handleRemoteReceiverOpen(receiver, connRequest);
         });
         connRequest.senderOpenHandler(sender -> {
-            HonoProtonHelper.setDefaultCloseHandler(sender);
-            // this should not happen -> no request-response for device clients.
-            LOG.debug("client [container: {}] wants to open a link [address: {}] for receiving messages",
-                    connRequest.getRemoteContainer(), sender.getRemoteSource());
-            sender.setCondition(ProtonHelper.condition(AmqpError.NOT_ALLOWED,
-                    "this adapter only forwards message to downstream applications"));
-            sender.close();
+            handleRemoteSenderOpen(sender, connRequest);
```

Comment:
As done above already:
`
// when an Attach frame with "role = receiver" is received

// when an Attach frame with "role = sender" is received

Would improve the (already good) documentation of the AMQP handling here IMHO.
`